replica/database.cc

```diff
@@ -1430,6 +1430,10 @@ void database::validate_new_keyspace(keyspace_metadata& ksm) {
             if (!_feat.keyspace_storage_options) {
                 throw exceptions::invalid_request_exception("Keyspace storage options not supported in the cluster");
             }
+            // It's non-system keyspace
+            if (!_user_sstables_manager->has_endpoint_client(so.endpoint)) {
```

Comment:
instead of requiring `database` to validate `storage_options`, probably it might be better if we can leave this to `sstable_manager` ?

it is already an "acquaintance" of `storage_options`, as it creates a new instance of `shared_sstable` with the specified `storage_option`.

Reply:
The point is to check endpoint presence when a keyspace is created, not when an sstable is created. The latter is _much_ later -- after a keyspace one would need to create a table, populate it with data and flush.

Reply:
yeah, i understand when this is called. i suggest to add a function in `sstable_manager` , and call that method instead. would be better to keep the `database` away from the internals of storage option verification.